doc/docs.md

```diff
@@ -781,22 +783,81 @@ type Alphabet = Abc | Xyz
 
 x := Alphabet(Abc{'test'}) // sum type
 if x is Abc {
-    // x is automatically cast to Abc and can be used here
+    // x is automatically casted to Abc and can be used here
     println(x)
 }
+if x !is Abc {
+    println('Not Abc')
+}
+```
+or using `match`:
+```v oksyntax
+match x {
+    Abc {
+        // x is automatically casted to Abc and can be used here
+        println(x)
+    }
+    Xyz {
+        // x is automatically casted to Xyz and can be used here
+        println(x)
+    }
+}
 ```
 
-If you have a struct field which should be checked, there is also a way to name an alias.
-```v ignore
-struct MyStruct {x int}
-struct MyStruct2 {y string}
+This works also with struct fields:
+```v
+struct MyStruct {
+    x int
+}
+struct MyStruct2 {
+    y string
+}
+
 type MySumType = MyStruct | MyStruct2
-struct Abc { bar MySumType }
-x := Abc{ bar: MyStruct{123} }
-if x.bar is MyStruct as bar {
-    // x.bar cannot be cast automatically
-    // you must explicitly state "as bar" to create a variable with the MyStruct type
-    println(bar)
+
+struct Abc {
+    bar MySumType
+}
+
+x := Abc{
+    bar: MyStruct{123} // MyStruct will be converted to MySumType type automatically
+}
+
+if x.bar is MyStruct {
+    // x.bar is automatically casted
+    println(x.bar)
+}
+
+match x.bar {
+    MyStruct {
+        // x.bar is automatically casted
+        println(x.bar)
+    }
+    else {}
+}
+```
+
+For mutable values, the compiler doesn't do a type cast by default. Mutable variables can change by the time and and doing a cast would be unsafe.
```

Comment:
Line is too long, should be wrapped.

Reply:
`Mutable variables can change, and doing a cast would be unsafe.` may be better.